discovery/addr.go

```diff
@@ -0,0 +1,83 @@
+package discovery
+
+import (
+	"fmt"
+	"net"
+	"sync"
+)
+
+type addrStatus bool
+
+const (
+	OK    = true
+	NotOK = false
+)
```

Comment:
I think without specifying the type, these will be plain ol' bools?

```suggestion
const (
	OK    addrStatus = true
	NotOK addrStatus = false
)
```